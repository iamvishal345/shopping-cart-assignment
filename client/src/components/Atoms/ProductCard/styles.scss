.product-card {
  background-color: white;
  padding: 0.5rem;
  box-shadow: 12px 0px 8px -10px #eeeeee;
  border-bottom: 2px dotted #eeeeee;
  .card-header {
    margin: 0;
    height: 60px;
    font-size: 15px;
  }
  .card-body {
    font-size: 12px;
    font-weight: 500;
    background-color: #eeeeee;
    padding: 6px 6px 8px;
    height: 60px;
    overflow: hidden;
  }
  .card-footer {
    display: flex;
    align-items: center;
    :nth-child(n) {
      flex: 1;
    }
  }
}
