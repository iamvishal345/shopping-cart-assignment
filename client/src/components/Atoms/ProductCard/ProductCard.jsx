import React from "react";
import "./styles.scss";

export const ProductCard = ({ product, handlePrimaryAction }) => {
  return (
    <div className="product-card">
      <h4 className="card-header">{product.name}</h4>
      <img
        loading="lazy"
        height="auto"
        width="100%"
        className="card-image"
        src={product.imageURL}
        alt={product.name}
      />
      <p title={product.description} className="card-body">
        {product.description}
      </p>
      <div className="card-footer">
        <h6>MRP Rs.{product.price}</h6>
        <button
          type="button"
          onClick={(e) => handlePrimaryAction(product)}
          className="btn"
        >
          Buy Now
        </button>
      </div>
    </div>
  );
};
