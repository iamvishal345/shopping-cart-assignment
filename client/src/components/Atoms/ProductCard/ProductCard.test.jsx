import React from "react";
import { shallow } from "enzyme";
import { ProductCard } from "./ProductCard";

const product = {
  name: "Fresho Kiwi - Green, 3 pcs",
  imageURL: "/static/images/products/fruit-n-veg/kiwi-green.jpg",
  description:
    "Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.",
  price: 87,
  stock: 50,
  category: "5b6899953d1a866534f516e2",
  sku: "fnw-kiwi-3",
  id: "5b6c6a7f01a7c38429530883",
};
const handlePrimaryAction = jest.fn();
const wrapped = shallow(
  <ProductCard product={product} handlePrimaryAction={handlePrimaryAction} />
);
describe("ProductCard", () => {
  it("should match the snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });
  it("should execute callback on click of buy now button", () => {
    wrapped.find(".btn").simulate("click", {});
    expect(handlePrimaryAction).toBeCalledWith(product);
  });
});
