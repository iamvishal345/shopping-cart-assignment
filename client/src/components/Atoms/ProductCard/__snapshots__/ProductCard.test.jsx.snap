// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should match the snapshot 1`] = `
<div
  className="product-card"
>
  <h4
    className="card-header"
  >
    Fresho Kiwi - Green, 3 pcs
  </h4>
  <img
    alt="Fresho Kiwi - Green, 3 pcs"
    className="card-image"
    height="auto"
    loading="lazy"
    src="/static/images/products/fruit-n-veg/kiwi-green.jpg"
    width="100%"
  />
  <p
    className="card-body"
    title="Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds."
  >
    Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds.
  </p>
  <div
    className="card-footer"
  >
    <h6>
      MRP Rs.
      87
    </h6>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Buy Now
    </button>
  </div>
</div>
`;
